tools/ci_build/github/apple/test_ios_packages.py

```diff
@@ -49,60 +50,59 @@ def _test_ios_packages(args):
         else:
             # If we specify the stage dir, then use it to create test project
-            stage_dir = args.test_project_stage_dir
+            stage_dir = args.test_project_stage_dir.resolve()
             if os.path.exists(stage_dir):
                 shutil.rmtree(stage_dir)
             os.makedirs(stage_dir)
 
-        # create a zip file contains the framework
-        # TODO, move this into a util function
-        local_pods_dir = os.path.join(stage_dir, 'local_pods')
-        os.makedirs(local_pods_dir, exist_ok=True)
-        # shutil.make_archive require target file as full path without extension
-        zip_base_filename = os.path.join(local_pods_dir, 'onnxruntime-mobile-c')
-        zip_file_path = zip_base_filename + '.zip'
-        shutil.make_archive(zip_base_filename, 'zip', root_dir=c_framework_dir, base_dir=framework_name)
+        # assemble the test project here
+        target_proj_path = stage_dir / 'ios_package_test'
 
-        # copy the test project to the temp_dir
-        test_proj_path = os.path.join(REPO_DIR, 'onnxruntime', 'test', 'platform', 'ios', 'ios_package_test')
-        target_proj_path = os.path.join(stage_dir, 'ios_package_test')
+        # copy the test project source files to target_proj_path
+        test_proj_path = pathlib.Path(REPO_DIR, 'onnxruntime/test/platform/ios/ios_package_test')
         shutil.copytree(test_proj_path, target_proj_path)
 
-        # generate the podspec file from the template
-        framework_info = load_framework_info(args.framework_info_file.resolve())
+        # assemble local pod files here
+        local_pods_dir = stage_dir / 'local_pods'
 
-        with open(os.path.join(REPO_DIR, 'VERSION_NUMBER')) as version_file:
-            ORT_VERSION = version_file.readline().strip()
+        # We will only publish xcframework, however, assembly of the xcframework is a post process
+        # and it cannot be done by CMake for now. See, https://gitlab.kitware.com/cmake/cmake/-/issues/21752
+        # For a single sysroot and arch built by build.py or cmake, we can only generate framework
+        # We still need a way to test it, replace the xcframework with framework in the podspec
+        framework_dir = args.c_framework_dir / framework_name
 
-        variable_substitutions = {
-            "VERSION": ORT_VERSION,
-            "IOS_DEPLOYMENT_TARGET": framework_info["IOS_DEPLOYMENT_TARGET"],
-            "WEAK_FRAMEWORK": framework_info["WEAK_FRAMEWORK"],
-            "LICENSE_FILE": '"LICENSE"',
-        }
+        pod_name, podspec = assemble_c_pod_package(staging_dir=local_pods_dir,
+                                                   pod_version=get_ort_version(),
+                                                   framework_info_file=args.framework_info_file,
+                                                   public_headers_dir=args.c_framework_dir / "Headers",
+                                                   framework_dir=framework_dir,
```

Comment:
You may need to set this to framework_dir / "Headers" if has_framework

Reply:
yep, just figured this out